### ISSUE TITLE
Date support

### ISSUE THREAD

**Issue by icanzilb** [icanzilb/MarkCodable#5]
Ideally the encoder/decoder should be able to write/read dates. The question is how to make those human friendly for editing? Obviously a fully-qualified date contains a lot of information which isn't quite "friendly" to edit by hand.

This seems like a good opportunity for adding a date strategy like for JSONEncoder and maybe add some predefined "human friendly" formats.

**Comment by DivineDominion** [comment]
Would suggest to start with ISO date formatter to at least get dates like `2022-09-07` (not `Date`s) right.
https://sarunw.com/posts/how-to-parse-iso8601-date-in-swift/

Date and time becomes tricky, though, because ISO timestamps are 🤮 

So if users enter `2022-09-07 15:45`, the parser should maybe produce a String instead of extracting a `Date` from `2022-09-07` ignoring the time, because that'd be pretty confusing.

**Comment by icanzilb** [comment]
Yeah, I have similar feelings about parsing dates ... not to mention that without a timezone a string date means nothing so I'm very cautious just adding this willie-nilly. I'm so burned by parsing dates that I ponder with the idea of just not having this feature at all but I can see how this might be something useful to have ...
